v3/src/components/graph/components/casedots.tsx

```diff
@@ -20,33 +20,36 @@ export const CaseDots = memo(function CaseDots(props: {
     graphModel = props.graphModel,
     dataset = useDataSetContext(),
     layout = useGraphLayoutContext(),
-    pointSizeMultiplier = graphModel.pointSizeMultiplier,
+    randomPointsRef = useRef<Record<string, { x: number, y: number }>>({}),
+    pointRadius = graphModel.getPointRadius(),
+    selectedPointRadius = graphModel.getPointRadius('select'),
     [dragID, setDragID] = useState(''),
     currPos = useRef({x: 0, y: 0}),
     target = useRef<any>(),
     xScale = layout.axisScale('bottom') as ScaleNumericBaseType,
     yScale = layout.axisScale('left') as ScaleNumericBaseType
 
-  const uniform = randomUniform(),
-    randomPointsRef = useRef<Record<string, { x: number, y: number }>>({})
-  dataset?.cases.forEach(({__id__}) => {
-    randomPointsRef.current[__id__] = {x: uniform(), y: uniform()}
-  })
+  useEffect(function initDistribution() {
+    const uniform = randomUniform()
+    dataset?.cases.forEach(({__id__}) => {
+      randomPointsRef.current[__id__] = {x: uniform(), y: uniform()}
+    })
+  }, [dataset?.cases])
```

Comment:
I agree that it should and am also puzzled if it doesn't. How can I set it up so that the useEffect only runs if there is a new dataset, but not when the cases change?